diff --git a/ansible/roles/keystone/tasks/config.yml b/ansible/roles/keystone/tasks/config.yml
index 38ac9de7..fce1b1d2 100644
--- a/ansible/roles/keystone/tasks/config.yml
+++ b/ansible/roles/keystone/tasks/config.yml
@@ -185,6 +185,26 @@
   notify:
     - Restart keystone-ssh container
 
+# NOTE(berendt): OIDC metadata directory integration {{{
+
+- name: Ensuring oidc metadata config directory exists
+  file:
+    path: "{{ node_config_directory }}/{{ item }}"
+    state: "directory"
+  with_items:
+    - "keystone/oidc_metadata"
+  when: inventory_hostname in groups['keystone']
+
+- name: Copy over oidc metadata files
+  copy:
+    src: "{{ item }}"
+    dest: "{{ node_config_directory }}/keystone/oidc_metadata"
+  with_fileglob:
+    - "{{ node_custom_config }}/keystone/oidc_metadata/*"
+  ignore_errors: true
+
+# }}}
+
 - name: Check keystone containers
   kolla_docker:
     action: "compare_container"
diff --git a/ansible/roles/keystone/templates/keystone.json.j2 b/ansible/roles/keystone/templates/keystone.json.j2
index c9415392..4bdfad00 100644
--- a/ansible/roles/keystone/templates/keystone.json.j2
+++ b/ansible/roles/keystone/templates/keystone.json.j2
@@ -1,4 +1,5 @@
 {% set keystone_cmd = 'apache2' if kolla_base_distro in ['ubuntu', 'debian'] else 'httpd' %}
+{% set keystone_oidc_metadata_dir = 'apache2/oidc_metadata' if kolla_base_distro in ['ubuntu', 'debian'] else 'httpd/oidc_metadata' %}
 {% set keystone_dir = 'apache2/conf-enabled' if kolla_base_distro in ['ubuntu', 'debian'] else 'httpd/conf.d' %}
 {
     "command": "/usr/sbin/{{ keystone_cmd }}",
@@ -35,6 +36,13 @@
             "dest": "/etc/{{ keystone_dir }}/wsgi-keystone.conf",
             "owner": "keystone",
             "perm": "0644"
+        },
+        {
+            "source": "{{ container_config_directory }}/oidc_metadata",
+            "dest": "/etc/{{ keystone_oidc_metadata_dir }}",
+            "owner": "keystone",
+            "perm": "0644",
+            "optional": true
         }
     ],
     "permissions": [
