diff --git a/ansible/roles/cinder/templates/cinder.conf.j2 b/ansible/roles/cinder/templates/cinder.conf.j2
index f409dc0a..8333a3b3 100644
--- a/ansible/roles/cinder/templates/cinder.conf.j2
+++ b/ansible/roles/cinder/templates/cinder.conf.j2
@@ -23,7 +23,8 @@ os_region_name = {{ openstack_region_name }}
 enabled_backends = {{ cinder_enabled_backends|map(attribute='name')|join(',') }}
 {% endif %}

-{% if service_name == "cinder-backup" and enable_ceph | bool and cinder_backend_ceph | bool %}
+{% if service_name == "cinder-backup" and enable_cinder_backup | bool %}
+{% if cinder_backup_driver == "ceph" %}
 backup_driver = cinder.backup.drivers.ceph
 backup_ceph_conf = /etc/ceph/ceph.conf
 backup_ceph_user = cinder-backup
@@ -46,6 +47,7 @@ backup_swift_auth_version = 1
 backup_swift_user =
 backup_swift_key =
 {% endif %}
+{% endif %}

 osapi_volume_listen = {{ api_interface_address }}
 osapi_volume_listen_port = {{ cinder_api_port }}
