diff --git a/ansible/roles/telegraf/templates/telegraf.conf.j2 b/ansible/roles/telegraf/templates/telegraf.conf.j2
index 667c7927..4f23a6c6 100644
--- a/ansible/roles/telegraf/templates/telegraf.conf.j2
+++ b/ansible/roles/telegraf/templates/telegraf.conf.j2
@@ -38,6 +38,8 @@
 [[inputs.kernel]]
 [[inputs.mem]]
 [[inputs.processes]]
+[[inputs.chrony]]
+  dns_lookup = false
 [[inputs.swap]]
 [[inputs.system]]
 [[inputs.net]]
